ansible/roles/csm.rr.ceph_zoning/tasks/main.yml

```diff
@@ -24,15 +24,40 @@
 ---
 # Apply CEPH zoning to make sure CEPH data gets replicated at rack level, 
 # so that there would not be data loss incase of a rack failure.
+
+- name: Check if CEPH rack entries with assigned children exist
+  shell: |
+    ceph osd tree -f json | jq -r '.nodes[] | select(.type == "rack" and (.children | length > 0)) | .name'
+  register: rack_check
+  delegate_to: "{{ play_hosts | first }}"
+  run_once: true
+  changed_when: false
+
+- name: Debug message when CEPH racks with children exist
+  when: rack_check.stdout_lines | length > 0
+  debug:
+    msg: "Racks already exist in CEPH OSD tree. Skipping CEPH zoning and CEPH haproxy reconfiguration"
+  run_once: true
+
+- name: Skipping CEPH zoning and CEPH haproxy reconfiguration since its already applied
+  meta: end_play
```

Comment:
This should not use `end_play`, for two reasons. 

1. This is not the last role in the playbook. `csm.rr.ceph_haproxy`  comes after it. Unless I am misunderstanding the check above, it is only checking to make sure the zone exists. It doesn't seem to be checking that the HA proxy config is done. And it shouldn't be, since that is an entirely different role, and that role should handle that. This role should use `when` conditionals to not do the remaining steps, rather than call `end_play`

2. Even if this did work fine currently, it can still cause problems if someone later needs to add a role after this one.

Reply:
After discussing with @sravani-sanigepalli , this task is going to be removed from the role, which will resolve this issue.